PolyEngine/Scripts/ProjectTool.py

```diff
@@ -0,0 +1,120 @@
+import argparse
+import os
+import shutil
+import sys
+import fileinput
+from enum import Enum
+
+class ActionType(Enum):
+    CREATE = 'Create',
+    UPDATE = 'Update'
+
+    def __str__(self):
+        return self.name
+
+# Custom action for project creation
+class CreateProjectAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs, **kwargs):
+        if nargs is not 2:
+            raise ValueError("Nargs must be 2 !")
+        super(CreateProjectAction, self).__init__(option_strings, dest, nargs, **kwargs)
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, ActionType.CREATE)
+        setattr(namespace, self.metavar[0], values[0])
+        setattr(namespace, self.metavar[1], values[1])
+
+class UpdateProjectAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs, **kwargs):
+        if nargs is not 1:
+            raise ValueError("Nargs must be 1 !")
+        super(UpdateProjectAction, self).__init__(option_strings, dest, nargs, **kwargs)
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, ActionType.UPDATE)
+        setattr(namespace, self.metavar, values[0])
+
+# Functions
+# Create project function
+
+# Setup:
+# $PROJECT_NAME$ - name of the project
+# $PROJECT_PATH$ - relative path of the project (game) directory (relative to root project dir)
+# $GAME_CLASS_NAME$ - name of the game class
+# $ENGINE_DIR$ - absolute path to engine root directory
+
+def ReplaceTagsInFile(fileToSearch, tagsAndValues):
+    print('Replacing tags in', fileToSearch)
+    for (tag, val) in tagsAndValues:
+        print('\tchanging', tag, 'to', val)
+
+    with fileinput.FileInput(fileToSearch, inplace=True) as file:
+        for line in file:
+            for (tag, val) in tagsAndValues:
+                line = line.replace(tag, val)
+            print(line, end='')
+
+def RunCmake(path, buildDir):
+    # Run cmake update (using undocumented parameters that work for some reason, src: http://cprieto.com/posts/2016/10/cmake-out-of-source-build.html)
+    if os.name == 'nt':
+        os.system(r'cmake -G "Visual Studio 14 2015 Win64" -H' + path + ' -B' + buildDir)
```

Comment:
Raw string is not necessary here. You should also use .format() to place variables in your string, for example:
`os.system('cmake -G "Visual Studio 14 2015 Win64" -H {} -B {}'.format(path, buildDir))`

Using .format() applies to all string concatenations in this file

Reply:
I'm using raw string to handle " inside of a string. But good point.

Reply:
You don't have to. If you start a string with single quotes you can use double quotes inside of this string as much as you like and it won't break. Of course it works the other way around too - if you need to use single quotes in your string, just start and end it with double quotes.

What is more, if you want to use ' and " inside your string, you still can do it starting your string with ''' (3 single quotes). In that case even \n won't break your string (this is a pythonic way to make multi-line comments btw).